java/src/jmri/jmrix/sprog/SprogTrafficController.java

```diff
@@ -182,7 +182,8 @@ public void sendSprogMessage(SprogMessage m) {
                 log.warn("sendMessage: no connection established");
             }
         } catch (Exception e) {
-            log.warn("sendMessage: Exception: {}", e.toString());
+//            log.warn("sendMessage: Exception: {}", e.toString());
+            log.warn("sendMessage: Exception: "+m.toString(), e);
```

Comment:
Concatenation should not be used in log messages (String concatenation is somewhat expensive since all String objects are essentially constants). It would be better to use the form ```log.warn("sendMessage: Exception: {}", m, e);``` (any ```.toString()``` in a logging argument just [slows down logging](https://www.slf4j.org/faq.html#logging_performance)).

Reply:
That was unintentional, corrected now.